StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/ElementDocumentationSummaryBase.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using StyleCop.Analyzers.Helpers;
+
+namespace StyleCop.Analyzers.DocumentationRules
+{
+    public abstract class ElementDocumentationSummaryBase : DiagnosticAnalyzer
+    {
+        abstract protected internal void HandleXmlElement(SyntaxNodeAnalysisContext context, XmlElementSyntax syntax, Location diagnosticLocation);
+        abstract protected internal void HandleXmlElement(SyntaxNodeAnalysisContext context, XmlElementSyntax syntax, Location[] diagnosticLocations);
```

Comment:
You can get away with just one `abstract` method here with a `params Location[] diagnosticLocations` parameter.